docs/guide/mocking-requests.md

```diff
@@ -0,0 +1,57 @@
+# Mocking Requests
+
+We write tests in order to be able to ship with confidence. The more underlying behavior we mock, the less confidence we get since the system is less closer to real usage.
+
+Mocking network requests is one of those tricky aspects of running tests in a non-browser environment. Web API's are not available so now we need to mock or install something that mocks this behavior.
+
+There are various tools for this job, but which one do you choose?
+
+We recommend [Mock Service Worker](https://mswjs.io/) to mock network requests. It handles both `REST` and `GraphQL` type of network requests and is framework agnostic.
```

Comment:
```suggestion
Because Vitest runs in Node, mocking network requests is tricky; web APIs are not available ,so we need something that will mimic network behavior for us. We recommend [Mock Service Worker](https://mswjs.io/) to accomplish this. It will let you mock both `REST` and `GraphQL` network requests, and is framework agnostic.
```

I think these four paragraphs can be condensed down to this:
* The first paragraph I think we can remove because it's a little generic about testing and mocking in general. If we want to include this, I think it probably belongs further "up" the documentation
* The third paragraph, posing a question to the reader, can be removed, too, as we're just going to answer it immediately.
* The second and fourth ones are combined